plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py

```diff
@@ -0,0 +1,130 @@
+    connect_yaml = f"""
+cluster_name: {cluster_name}
+provider:
+  type: aws
+  region: us-west-2
+  availability_zone: us-west-2a,us-west-2b
+  cache_stopped_nodes: False
+  key_pair:
+     key_name: hydra_test_{cluster_name}
+auth:
+  ssh_user: ubuntu
+head_node:
+  InstanceType: m5.large
+  ImageId: ami-008d8ed4bd7dc2485
+    """
+    with tempfile.NamedTemporaryFile(suffix=".yaml", delete=False) as f:
+        with open(f.name, "w") as file:
+            print(connect_yaml, file=file)
+        temp_yaml = f.name
+        ray_up(temp_yaml)
+        ray_new_dir(temp_yaml, temp_remote_dir, False)
+        yield
+        ray_down(f.name)
+
+
+@pytest.mark.skipif(sys.platform.startswith("win"), reason=win_msg)
+@pytest.mark.usefixtures("manage_cluster")
+@pytest.mark.parametrize(
+    "launcher_name, overrides, tmpdir",
+    [
+        (
+            "ray_aws",
+            [
+                f"hydra.launcher.params.ray_cluster_cfg.cluster_name={cluster_name}",
+                "hydra.launcher.params.stop_cluster=False",
+                f"hydra.launcher.params.sync_down.source_dir={sweep_dir}/",
+                f"hydra.launcher.params.sync_down.target_dir={sweep_dir}",
+                f"hydra.launcher.params.ray_cluster_cfg.provider.key_pair.key_name=hydra_test_{cluster_name}",
+            ],
+            Path(sweep_dir),
+        )
+    ],
+)
+class TestRayAWSLauncher(LauncherTestSuite):
+    pass
+
+
+@pytest.mark.skipif(sys.platform.startswith("win"), reason=win_msg)
+@pytest.mark.usefixtures("manage_cluster")
+@pytest.mark.parametrize(
+    "tmpdir,  task_launcher_cfg, extra_flags",
+    [
+        (
+            Path(temp_remote_dir),
+            {
+                "defaults": [
+                    {"hydra/launcher": "ray_aws"},
+                    {"hydra/hydra_logging": "hydra_debug"},
+                    {"hydra/job_logging": "disabled"},
+                ]
+            },
+            [
+                "-m",
+                f"hydra.launcher.params.ray_cluster_cfg.cluster_name={cluster_name}",
+                "hydra.launcher.params.stop_cluster=False",
+                f"hydra.launcher.params.sync_down.source_dir={temp_remote_dir}/",
+                f"hydra.launcher.params.sync_down.target_dir={temp_remote_dir}",
+                f"hydra.launcher.params.ray_cluster_cfg.provider.key_pair.key_name=hydra_test_{cluster_name}",
+                f"hydra.launcher.params.ray_cluster_cfg.provider.key_pair.key_name=hydra_test_{cluster_name}",
+            ],
+        )
+    ],
+)
+class TestRayAWSLauncherIntegration(IntegrationTestSuite):
+    """
+    Run this launcher through the integration test suite.
+    """
+
+    def get_test_app_working_dir(self) -> Optional[Path]:
+        """
+        By default test applications working dir is tmpdir, override this method if that's not the case.
+        This could be helpful when the tests kick off applications on remote machines.
+        """
+        return Path("/home/ubuntu")
```

Comment:
This seems awfully specific to the instance types you get on aws in the test.
not sure if it's an issue or not.

Reply:
Yes you are right. Since we pinged the AMI type in the test, i think this is fine for now.